### PR TITLE
Added PK clear command and online command.

### PR DESCRIPTION
**Pull request by R3DIANCE**

### CODE REVIEW

File: src/DataModel/Configuration/GameConfiguration.cs [hunk 1]
```diff
@@ -22,6 +22,11 @@ public class GameConfiguration
         /// </summary>
         public short MaximumLevel { get; set; }
 
+        /// <summary>
+        /// Gets or sets the zen cost of PK Clear command.
+        /// </summary>
+        public int PKClearZenCost { get; set; }
```
Comment (sven-n) [inline src/DataModel/Configuration/GameConfiguration.cs:28]: Can you replace the word "Zen" by "Money"?

File: src/GameLogic/PlugIns/Addons/AutoReconnect.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    using MUnique.OpenMU.PlugIns;
+
+    [Guid("8336259B-4EC6-4280-A4F6-549F0A967F59")]
+    [PlugIn("Auto reconnection", "Automaticly reconnected to game in case of player lost disconnecting.")]
+
+    public class AutoReconnect
+    {
+        // logic here
+    }
+}
```
Comment (sven-n) [inline src/GameLogic/PlugIns/Addons/AutoReconnect.cs:18]: please remove it for this PR

File: src/GameLogic/PlugIns/Addons/PlayerEnterGameMessage.cs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    using MUnique.OpenMU.PlugIns;
+
+    [Guid("E139CC14-1558-4FF7-8F1E-87773E6C838E")]
+    [PlugIn("Enter Game Message", "Displays a message when player connected.")]
+
+    public class PlayerEnterGameMessage
+    {
+        // logic here
+    }
+}
```
Comment (sven-n) [inline src/GameLogic/PlugIns/Addons/PlayerEnterGameMessage.cs:18]: please remove it for this PR

File: src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+        public CharacterStatus MinCharacterStatusRequirement => CharacterStatus.Normal;
+
+        /// <inheritdoc />
+        public void HandleCommand(Player player, string command)
+        {
+            try
+            {
+                int Cost = player.GameContext.Configuration.PKClearZenCost;
+
+                if ((int)player.SelectedCharacter.State >= 4 && player.Money >= Cost)
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs:36]: You can use the HeroState enum to compare

File: src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+        /// <inheritdoc />
+        public CharacterStatus MinCharacterStatusRequirement => CharacterStatus.Normal;
+
+        /// <inheritdoc />
+        public void HandleCommand(Player player, string command)
+        {
+            try
+            {
+                int Cost = player.GameContext.Configuration.PKClearZenCost;
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs:34]: Cost -> cost

File: src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+                    if (Cost <= 0)
+                    {
+                        player.ViewPlugIns.GetPlugIn<IUpdateCharacterHeroStatePlugIn>()?.UpdateCharacterHeroState();
+                        player.ShowMessage($"[PK System] Status cleard.");
+                        return;
+                    }
+
+                    player.TryRemoveMoney(Cost);
+                    player.ViewPlugIns.GetPlugIn<IUpdateCharacterHeroStatePlugIn>()?.UpdateCharacterHeroState();
+                    player.ShowMessage($"[PK System] Status cleard, -{Cost} zen.");
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs:49]: TryRemoveMoney returns a bool value, if it was successful. Only then the State should be changed to Normal.

File: src/GameLogic/PlugIns/ChatCommands/Commands/PartyCommand.cs [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+                // logic here
+            }
+            catch (ArgumentException e)
+            {
+                player.ShowMessage(e.Message);
+            }
+        }
+
+    }
+}
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PartyCommand.cs:46]: Please remove for this PR

File: src/Persistence/EntityFramework/Migrations/00000000000000_Initial.cs [hunk 8]
```diff
@@ -70,6 +70,7 @@ protected override void Up(MigrationBuilder migrationBuilder)
                 {
                     Id = table.Column<Guid>(nullable: false),
                     MaximumLevel = table.Column<short>(nullable: false),
+                    PKClearZenCost = table.Column<int>(nullable: false, defaultValueSql: "300000"),
```
Comment (sven-n) [inline src/Persistence/EntityFramework/Migrations/00000000000000_Initial.cs:73]: Please remove the defaultValueSql. Instead, add the assignment of the initial value in the DataInitialization.InitializeGameConfiguration() function.

There are also more places where it needs to be added:
* EntityDataContextModelSnapshot
* 00000000000000_Initial.designer.cs

File: src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                int Cost = player.GameContext.Configuration.PKClearZenCost;
+
+                if ((int)player.SelectedCharacter.State >= 4 && player.Money >= Cost)
+                {
+                    player.SelectedCharacter.State = HeroState.Normal;
+
+                    if (Cost <= 0)
+                    {
+                        player.ViewPlugIns.GetPlugIn<IUpdateCharacterHeroStatePlugIn>()?.UpdateCharacterHeroState();
+                        player.ShowMessage($"[PK System] Status cleard.");
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs:43]: cleard -> cleared

File: src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                else
+                {
+                    if ((int)player.SelectedCharacter.State <= 3 && player.Money >= Cost)
+                    {
+                        player.ShowMessage($"[PK System] Status cannot be cleard, your status is {player.SelectedCharacter.State.ToString().ToLower()}.");
+                    }
+                    else if ((int)player.SelectedCharacter.State >= 4 && player.Money < Cost)
+                    {
+
+                        player.ShowMessage($"[PK System] Status cannot be cleard, zen reqiuerd: {Cost}.");
```
Comment (sven-n) [inline src/GameLogic/PlugIns/ChatCommands/Commands/PKClearCommand.cs:60]: reqiuerd -> required

**Comment by R3DIANCE** [comment]
yessir!

**Pull request closed without merging**